src/App.tsx

```diff
@@ -72,13 +84,21 @@ const App = () => {
   const { connectOnLoad } = useConnectionStore();
   const { resetJetton } = useJettonLogo();
   const location = useLocation();
+  const { isAddressEmpty, jettonAddress } = useJettonAddress();
+  const { showNotification } = useNotification();
 
   useEffect(() => {
     resetJetton();
   }, [location.pathname]);
 
   useEffect(() => {
     connectOnLoad();
+    const isIncorrectAddress = !isAddressEmpty && !jettonAddress;
+    const isEmptyAddress =
+      (!jettonAddress && location.pathname === ROUTES.jetton + "/") ||
+      location.pathname === ROUTES.jetton;
```

Comment:
how can this happen? I'd expect react router to redirect to the page not found route in this case